### PR TITLE
Preserve the binary jar name in a prebuilt_jar

### PR DESCRIPTION
**Pull request by shs96c**
This makes it far easier to do things like build release zips
from artifacts of the build for OSS projects.

### CODE REVIEW

**Comment by marcinkosiba** [comment]
where are the tests?

**Comment by facebook-github-bot-7** [comment]
By analyzing the blame information on this pull request, we identified @Coneko, @andrewjcg and @mikekap to be potential reviewers.

**Comment by facebook-github-bot-7** [comment]
Thank you for your pull request.  As you may know, we require contributors to sign our Contributor License Agreement, and we don't seem to have you on file and listed as active anymore.  In order for us to review and merge your code, please email cla@fb.com with your details so we can update your status.

File: src/com/facebook/buck/jvm/java/PrebuiltJar.java [hunk 1]
```diff
@@ -225,16 +229,18 @@ public BuildableProperties getProperties() {
...
-      steps.add(new MakeCleanDirectoryStep(getProjectFilesystem(), copiedBinaryJar));
-      steps.add(CopyStep.forDirectory(
-          getProjectFilesystem(),
-          resolvedBinaryJar,
-          copiedBinaryJar,
-          CopyStep.DirectoryMode.CONTENTS_ONLY));
-    } else {
-      steps.add(new MkdirStep(getProjectFilesystem(), copiedBinaryJar.getParent()));
-      steps.add(CopyStep.forFile(getProjectFilesystem(), resolvedBinaryJar, copiedBinaryJar));
+    if (!resolvedBinaryJar.equals(getProjectFilesystem().resolve(copiedBinaryJar))) {
```
Comment (marcinkosiba) [inline src/com/facebook/buck/jvm/java/PrebuiltJar.java]: when is this true?

Comment (shs96c) [inline src/com/facebook/buck/jvm/java/PrebuiltJar.java]: Probably never, unless there's been a terrible foul up.

**Comment by shs96c** [comment]
There are no tests.

**Comment by facebook-github-bot-6** [comment]
@shs96c updated the pull request.

**Comment by shs96c** [comment]
No idea why `//test/com/facebook/buck/android:integration` fails in Travis. Works fine for me (tm)

**Comment by facebook-github-bot-0** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

**Comment by marcinkosiba** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**